src/invoker_cache.cpp

```diff
@@ -72,6 +72,32 @@ boost::optional<const Invoker&> InvokerCache::GetFound1_0(const std::string& net
     return invoker->second;
 }
 
+boost::optional<const std::string&> InvokerCache::GetFound1_0Id(const std::string& network_config,
```

Comment:
This name is hardly understandable. Can we use something like GetFound1_0SolverName?

Reply:
Would you be okay with GetFound1_0SolverId?

Reply:
Of course

Reply:
But please note that I think that distinguishing the difference between _numeric_ and _string_ solver identifiers is important because strings are much more expensive.

I still recommend using "ID" for the numeric identifiers, and "Name" for the strings.

Reply:
I don't think there is much overhead, but I'd rather have them both replaced a bit later everywhere we can by `solver::Id`. That is as fast as number (as it is one), type-safe and trivially and cheaply enough (tiny map fetch) converted to string.